src/main/java/at/helpch/placeholderapi/expansion/server/util/ServerUtil.java

```diff
@@ -0,0 +1,158 @@
+package at.helpch.placeholderapi.expansion.server.util;
+
+import org.bukkit.Bukkit;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.TreeMap;
+
+public final class ServerUtil {
+
+    private static final Map<String, String> variants = new TreeMap<>();
+    private static final String variant;
+    private static final String serverVersion;
+    private static final String build;
+
+    // TPS stuff
+    private static final Object craftServer;
+    private static final Field tpsField;
+    private static boolean hasTpsMethod; // Paper and its forks have Bukkit#getTps
+
+    static {
+        variants.put("net.pl3x.purpur.PurpurConfig", "Purpur");
+        variants.put("gg.airplane.AirplaneConfig", "Airplane");
+        variants.put("com.tuinity.tuinity.config.TuinityConfig", "Tuinity");
+        variants.put("io.papermc.paper.configuration.ConfigurationLoaders", "Paper"); // New config location for Paper 1.19+
+        variants.put("com.destroystokyo.paper.PaperConfig", "Paper"); // Still supported by Paper, but deprecated.
+
+        craftServer = getCraftServer();
+        tpsField = getTpsHandler();
+        variant = findVariant();
+        serverVersion = Bukkit.getBukkitVersion().split("-")[0]; // The version is formatted as 1.20.1-R0.1-SNAPSHOT
+        build = findBuild();
+    }
+
+    private static String findVariant() {
+        try {
+            for (final Map.Entry<String, String> entry : variants.entrySet()) {
+                Class.forName(entry.getKey());
+                return entry.getValue();
+            }
```

Comment:
You sure this won't cancel itself out should forName throw?

Like if it tries first entry and it fails, what makes this keep going in the for loop and not just cancel it completely?